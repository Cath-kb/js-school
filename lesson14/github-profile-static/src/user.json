{
  "login": "Cath-kb",
  "id": 2164575,
  "node_id": "MDQ6VXNlcjIxNjQ1NzU=",
  "avatar_url": "https://avatars3.githubusercontent.com/u/2164575?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/Cath-kb",
  "html_url": "https://github.com/Cath-kb",
  "followers_url": "https://api.github.com/users/Cath-kb/followers",
  "following_url": "https://api.github.com/users/Cath-kb/following{/other_user}",
  "gists_url": "https://api.github.com/users/Cath-kb/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/Cath-kb/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/Cath-kb/subscriptions",
  "organizations_url": "https://api.github.com/users/Cath-kb/orgs",
  "repos_url": "https://api.github.com/users/Cath-kb/repos",
  "events_url": "https://api.github.com/users/Cath-kb/events{/privacy}",
  "received_events_url": "https://api.github.com/users/Cath-kb/received_events",
  "type": "User",
  "site_admin": false,
  "name": "Catherine Chernysheva",
  "company": null,
  "blog": "https://www.linkedin.com/in/cathkb",
  "location": "Kyiv, Ukraine",
  "email": null,
  "hireable": null,
  "bio": null,
  "public_repos": 21,
  "public_gists": 1,
  "followers": 4,
  "following": 0,
  "created_at": "2012-08-16T17:44:34Z",
  "updated_at": "2018-08-05T08:14:18Z"
}
