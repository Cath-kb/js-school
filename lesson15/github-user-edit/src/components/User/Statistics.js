import React from "react";

import { STATISTIC_TITLES as titles } from '../../consts';

const StatisticItem = ({title, value}) => (
  <li className="list-group-item d-flex justify-content-between align-items-center">
    {title}: <span className="badge badge-primary badge-pill">{value}</span>
  </li>
);

const Statistics = ({data}) => {
  const keys = Object.keys(data);

  return (
    <React.Fragment>
      <h4><strong>Statistics:</strong></h4>
      <ul className="list-group">
        {keys.map(key => (
          <StatisticItem
            key={key}
            title={titles[key] || key}
            value={data[key]}
          />
        ))}
      </ul>
    </React.Fragment>
  );
};

export default Statistics;
