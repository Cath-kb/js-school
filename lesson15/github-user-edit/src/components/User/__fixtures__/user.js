const user = {
  avatar: {
    url: 'https://avatars3.githubusercontent.com/u/2164575?v=4',
  },
  person: {
    name: 'John Doe',
    login: 'joe',
    company: '',
    location: 'Aldebaran',
  },
  statistics: {
    repos: 5,
    following: 0,
    followers: 5,
  }
};

export default user;