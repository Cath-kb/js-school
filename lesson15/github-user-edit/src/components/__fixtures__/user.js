const user = {
  avatar: {
    url: 'https://avatars3.githubusercontent.com/u/2164575?v=4',
  },
  person: {
    name: 'John Doe',
    login: 'joe',
    company: '',
    location: 'Aldebaran',
    origin: 'https://github.com/joe',
  },
  statistics: {
    repos: 5,
    following: 0,
    followers: 5,
  }
};

export default user;