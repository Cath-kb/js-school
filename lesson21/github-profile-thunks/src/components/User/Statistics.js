import React from "react";
const titles = {
  repos: 'Repositories',
  following: 'Following',
  followers: 'Followers',
};

const StatisticItem = ({title, value}) => (
  <li className="list-group-item d-flex justify-content-between align-items-center">
    {title}: <span className="badge badge-primary badge-pill">{value}</span>
  </li>
);

const Statistics = ({data}) => {
  const keys = Object.keys(data);

  return (
    <div>
      <h4><strong>Statistics:</strong></h4>
      <ul className="list-group">
        {keys.map(key => (
          <StatisticItem
            key={key}
            title={titles[key] || key}
            value={data[key]}
          />
        ))}
      </ul>
    </div>
  );
};

export default Statistics;
