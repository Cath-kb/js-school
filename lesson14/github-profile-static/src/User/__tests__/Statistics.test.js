import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import Statistics from './../Statistics';
import user from './../__fixtures__/user';

const { statistics } = user;

describe('Statistics', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Statistics data={statistics} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be rendered without errors', () => {
    const html = renderer.create(<Statistics data={statistics} />).toJSON();
    expect(html).toMatchSnapshot();
  });

  it('should be 3 items', () => {
    const html = renderer.create(<Statistics data={statistics} />);
    const items = html.root.findAllByType('li');
    expect(items).toHaveLength(3);
  });
});
