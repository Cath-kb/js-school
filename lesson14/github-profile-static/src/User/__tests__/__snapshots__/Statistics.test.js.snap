// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics should be rendered without errors 1`] = `
<div>
  <h4>
    <strong>
      Statistics:
    </strong>
  </h4>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Repositories
      : 
      <span
        className="badge badge-primary badge-pill"
      >
        5
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Following
      : 
      <span
        className="badge badge-primary badge-pill"
      >
        0
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Followers
      : 
      <span
        className="badge badge-primary badge-pill"
      >
        5
      </span>
    </li>
  </ul>
</div>
`;
